### ISSUE TITLE
Pull replication after database delete yields "Caught up with changes!" 

### ISSUE THREAD

**Issue by PiranhaTodd** [couchbase/couchbase-lite-ios#169]
Setup:  Create a new database on a remote server using whatever means you like.  Add a document or two, and add a property or two to each document.

Repro:
1) In a CBL-iOS app, create a blank local database and use the same name as the database you created on the server above.  Do this with CBLManager createDatabaseNamed.
2) Create a pull replication from the remote server, using the database you just created.  Do this with CBLDatabase replicateFromURL.
3) I'm not sure if this matters, but for the resulting CBLReplication object, we set the 'continuous' property to YES.  We do not set the 'persistent' property (I assume that means NO by default).
4) After starting the replication, things proceed as expected, and the result is that the database now exists on the local iOS device with all documents and properties as expected.
5) Once the pull replication is complete, 'stop' it, and then destroy the CBLReplication object - let it go out of scope.
6) Delete the local database via CBLDatabase deleteDatabase.
7) Repeat steps 1-3.

Result:
After starting the replication, you'll see something like the output below.  Note the "Received 0 changes" and "Caught up with changes!" logs.

Expected:
Since this is a brand new database, I'd expect the new pull replication to, um, pull stuff.  :)  But CBL acts like nothing needs to be pulled.

```
16:05:01.932‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338] STARTING ...
16:05:01.932‖ SyncVerbose: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Found credential, using CBLBasicAuthorizer[anyurinateredisilessedde/****]
16:05:01.940‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Reachability state = <hammersmith-stage.cloudant.com>:unreachable (00)
16:05:01.944‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Reachability state = <hammersmith-stage.cloudant.com>:reachable (02)
16:05:01.944‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Going online
16:05:01.945‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: postProgressChanged (0/0, active=1 (batch=0, net=1), online=1)
16:05:01.945‖ SyncVerbose: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: GET _local/12083e47ad3b2a3cb6fd0ee3f4fd2e2aa33529a1
16:05:01.946‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: postProgressChanged (0/0, active=1 (batch=0, net=1), online=1)
16:05:02.079‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Replicating from lastSequence=37-g1AAAADneJzLYWBgYMlgTmFQSElKzi9KdUhJMtJLTMlPSjUwMNRLzskvTUnMK9HLSy3JAapkSmRIsv___39WIjvxepIcgGRSPVgbD_Ha8liAJEMDkALq3J-VyEmq1gMQrUBbObMA_RpKVw
16:05:02.079‖ SyncVerbose: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338] starting ChangeTracker: mode=0, since=37-g1AAAADneJzLYWBgYMlgTmFQSElKzi9KdUhJMtJLTMlPSjUwMNRLzskvTUnMK9HLSy3JAapkSmRIsv___39WIjvxepIcgGRSPVgbD_Ha8liAJEMDkALq3J-VyEmq1gMQrUBbObMA_RpKVw
16:05:02.081‖ SyncVerbose: CBLSocketChangeTracker[0x14d1ce70 proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: GET //anyurinateredisilessedde:QwhbSydT1cICOtxBKiWHvTN5@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338/_changes?feed=normal&heartbeat=300000&style=all_docs&since=37-g1AAAADneJzLYWBgYMlgTmFQSElKzi9KdUhJMtJLTMlPSjUwMNRLzskvTUnMK9HLSy3JAapkSmRIsv___39WIjvxepIcgGRSPVgbD_Ha8liAJEMDkALq3J-VyEmq1gMQrUBbObMA_RpKVw&limit=100
16:05:02.409‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Received 0 changes
16:05:02.410‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: postProgressChanged (0/0, active=1 (batch=0, net=1), online=1)
16:05:02.410‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: Caught up with changes!
16:05:02.410‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: postProgressChanged (0/0, active=0 (batch=0, net=0), online=1)
16:05:02.413‖ SyncVerbose: CBLSocketChangeTracker[0x14d1ce70 proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: GET //anyurinateredisilessedde:QwhbSydT1cICOtxBKiWHvTN5@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338/_changes?feed=longpoll&heartbeat=300000&style=all_docs&since=37-g1AAAADneJzLYWBgYMlgTmFQSElKzi9KdUhJMtJLTMlPSjUwMNRLzskvTUnMK9HLSy3JAapkSmRIsv___39WIjvxepIcgGRSPVgbD_Ha8liAJEMDkALq3J-VyEmq1gMQrUBbObMA_RpKVw&limit=100
16:05:02.422‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338] STOPPING...
16:05:02.422‖ Sync: Stopping 0 remote requests
16:05:02.422‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338] STOPPED
16:05:02.423‖ Replication: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338] took 0.490 sec; error=(null)
16:05:02.424‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: postProgressChanged (0/0, active=0 (batch=0, net=0), online=1)
16:05:02.424‖ Sync: CBL_Puller[https://anyurinateredisilessedde:*****@hammersmith-stage.cloudant.com/proj_551b0190-9adb-4c71-bd8d-bbb040b53338]: postProgressChanged (0/0, active=0 (batch=0, net=0), online=1)
```
